docs/vnext/openapi.yaml

```diff
@@ -0,0 +1,405 @@
+        format: uuid
+        example: 00000000-0000-0000-0000-000000000000
+    HeaderCorrelationId:
+      name: x-ms-correlation-request-id
+      in: header
+      required: true
+      description: Client or ARM generated GUID identifying a set of related operations. Should be included in logs and traces.
+      schema:
+        type: string
+        format: uuid
+        example: 00000000-0000-0000-0000-000000000000
+    HeaderTraceParent:
+      name: traceparent
+      required: true
+      description: Standard W3C trace context header generated by OpenTelemetry. Providers should use the header if they support OpenTelemetry.
+      in: header
+      schema:
+        type: string
+    HeaderTraceState:
+      name: tracestate
+      required: true
+      description: Standard W3C trace context header generated by OpenTelemetry. Providers should use the header if they support OpenTelemetry.
+      in: header
+      schema:
+        type: string
+    ProviderVersion:
+      name: providerVersion
+      in: path
+      required: true
+      description: Version of the provider. This is specified by the user with the provider import syntax in the Bicep or ARM template file. It is recommended for the extensibility provider to validate the request payload against the provider version, but it's not mandatory.
+      schema:
+        type: string
+    ReferenceId:
+      name: referenceId
+      in: path
+      required: true
+      description: Reference ID of the resource.
+      schema:
+        type: string
+    ResourceOperationPath:
+      name: resourceOperationPath
+      in: path
+      required: true
+      description: relative path to the resource operation.
+      schema:
+        type: string
+  headers:
+    Date:
+      description: The date that the request was processed, in [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.1.2) format.
+      schema:
+        type: string
+        format: date
+  responses:
+    Accepted:
+      description: Accepted
+      headers:
+        Date:
+          $ref: '#/components/headers/Date'
+        Location:
+          description: An absolute and public facing URI pointing to an operation resource which can be used to monitor the status of the delete operation.
+          schema:
+            type: string
+        Retry-After:
+          description: Indicates how long the client should wait before making a follow-up request. The value must only be specified in seconds, with a minimum of 10 seconds and a maximum of 10 minutes.
+          schema:
+            type: integer
+    BadRequest:
+      description: Failed operation
+      headers:
+        Date:
+          $ref: '#/components/headers/Date'
+      content: 
+        application/json: 
+          schema:
+            $ref: '#/components/schemas/ErrorResponse'
+    ResourceResponse:
+      description: Succesful operation
+      headers:
+        Date:
+          $ref: '#/components/headers/Date'

```

Comment:
Should we support a version header in the response? (Could include it in our logs for investigations.)

Reply:
Good point. Added `x-ms-request-id` to make it consistent with ARM RPC.